From 3897dc21683672458965da582fb8de833d925f27 Mon Sep 17 00:00:00 2001
From: Vladimir Oltean <olteanv@gmail.com>
Date: Tue, 4 Sep 2018 12:16:00 +0300
Subject: [PATCH 1/2] sepolicy: public: add TCSETSF to the list of unprivileged
 TTY ioctls

* This allows interactive "su" sessions started from a terminal app
  (therefore running in untrusted_app context) to fully interact with
  their respective untrusted_app_devpts.
* This fixes "su" behavior such as broken tab completion, or
  Ctrl-C exiting the su shell.
* Denial:

  avc: denied { ioctl } for pid=5629 comm="su" path="/dev/pts/2" dev="devpts"
  ino=5 ioctlcmd=5404 scontext=u:r:untrusted_app:s0:c512,c768
  tcontext=u:object_r:untrusted_app_devpts:s0:c512,c768
  tclass=chr_file permissive=0

Change-Id: Ic0867a40ed29348abb03bb91df4c5807379f9df0
Signed-off-by: Vladimir Oltean <olteanv@gmail.com>
---
 public/ioctl_macros | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/public/ioctl_macros b/public/ioctl_macros
index f7081d57..5d43ae2d 100644
--- a/public/ioctl_macros
+++ b/public/ioctl_macros
@@ -49,7 +49,7 @@ define(`unpriv_unix_sock_ioctls', `{
 # commonly used TTY ioctls
 # merge with unpriv_unix_sock_ioctls?
 define(`unpriv_tty_ioctls', `{
-  TIOCOUTQ FIOCLEX TCGETS TCSETS TIOCGWINSZ TIOCSWINSZ TIOCSCTTY TCSETSW
+  TIOCOUTQ FIOCLEX TCGETS TCSETS TCSETSF TIOCGWINSZ TIOCSWINSZ TIOCSCTTY TCSETSW
   TCFLSH TIOCSPGRP TIOCGPGRP
 }')
 
-- 
2.17.1

